### PR TITLE
migration article work

### LINKED ISSUE #22858: Clean up migration area of .NET guidance
The migration/porting information has a lot of irrelevant information, such as migrating from [DNX (RC1 of 1.0) to .NET CLI (RC2 of 1.0)](https://docs.microsoft.com/en-us/dotnet/core/migration/from-dnx). All of this old information should be deleted and/or brought up-to-date.

This issue is here to track that effort.

**Migration guide**
- Migration
  - [x] [.NET Core 2.0 to 2.1](https://docs.microsoft.com/en-us/dotnet/core/migration/20-21) (**DELETE** but move `<DotNetCliToolReference>` info to the [Breaking Changes 2.1 article](https://docs.microsoft.com/en-us/dotnet/core/compatibility/2.1)) (#22917)
  - [x] [Migrate from project.json](https://docs.microsoft.com/en-us/dotnet/core/migration/index) (**DELETE** project.json/.xproj/DNX material hasn't applied for years.) (#22919)
  - [x] [Map between project.json and csproj](https://docs.microsoft.com/en-us/dotnet/core/tools/project-json-to-csproj) (**DELETE**) (#22919)
  - [x] [Map AssemblyInfo attributes to properties](https://docs.microsoft.com/en-us/dotnet/core/migration/assembly-info) (**MOVE** to the MSBuild info area) (#22918)
  - [x] [Changes in CLI overview](https://docs.microsoft.com/en-us/dotnet/core/tools/cli-msbuild-architecture) (**DELETE** conceptual material describing pre .NET Core 1 world) (#22919)
  - [x] [Migrate from DNX](https://docs.microsoft.com/en-us/dotnet/core/migration/from-dnx) (**DELETE** DNX material hasn't applied for years) (#22919)

- Port from .NET Framework
  - [x] [Overview](https://docs.microsoft.com/en-us/dotnet/core/porting/index) (**UPDATE/SPLIT** Article isn't really an overview. Lots of information, some steps to follow too, article has a mixed identity) (#23218)
  - [ ] [Analyze third-party dependencies](https://docs.microsoft.com/en-us/dotnet/core/porting/third-party-deps) (**UPDATE** Need better intro and flow)
  - [x] [Port libraries](https://docs.microsoft.com/en-us/dotnet/core/porting/libraries) (**UPDATE** Good information, should be pointed to by other articles such as port WPF) (#23218)
  - [x] [Organize projects for .NET](https://docs.microsoft.com/en-us/dotnet/core/porting/project-structure) (**INVESTIGATE** Describes code sharing. Probably more useful back when WinForms/WPF wasn't available. Not sure how relevant this is anymore) (#23218)
  - [x] [Unavailable technologies](https://docs.microsoft.com/en-us/dotnet/core/porting/net-framework-tech-unavailable?toc=/dotnet/fundamentals/toc.json&bc=/dotnet/breadcrumb/toc.json) (**KEEP**) (#23218)
  - [x] [Tools](https://docs.microsoft.com/en-us/dotnet/core/porting/tools) (**DELETE** Too small. Move content to the overview article) (#23218)
  - [x] [Use the Windows Compatibility Pack](https://docs.microsoft.com/en-us/dotnet/core/porting/windows-compat-pack) (**UPDATE** Good information, should be pointed to by other articles such as port WPF) (#23218)
  - [x] [Port Windows Forms projects](https://docs.microsoft.com/en-us/dotnet/desktop/winforms/migration/?view=netdesktop-5.0) (**KEEP**) (#23218)
  - [x] [Port WPF projects](https://docs.microsoft.com/en-us/dotnet/desktop/wpf/migration/convert-project-from-net-framework) (**KEEP** Link is redirecting, fix) (#23218)
  - [x] [Port C++/CLI projects](https://docs.microsoft.com/en-us/dotnet/core/porting/cpp-cli) (**KEEP**) (#23218)

Possible future

- [x] Reference MSBuild Assembly Info attribute section in future content
- [x] File issue for MSBuild Assembly Info section to break out all of the individual settings

### PR DESCRIPTION
**Pull request by adegeo**
## Summary

- Rewrote the overview article.
- Worked on acro for other articles.
- Updated commands for upgrade tool.
- Deleted tools.md file and moved content into overview.
- Changed a lot of .NET Core references to .NET.

Fixes #22858

[Internal preview](https://review.docs.microsoft.com/en-us/dotnet/core/porting/?branch=pr-en-us-23218)

### CODE REVIEW

File: docs/core/porting/index.md [hunk 1]
```diff
@@ -1,116 +1,164 @@
...
+For more information, see [.NET API analyzer](../../standard/analyzers/api-analyzer.md).
 
-1. At this point, you can switch to targeting .NET Core (generally for applications) or .NET Standard (for libraries).
+## Considerations when porting
 
-   The choice between .NET Core and .NET Standard is largely dependent on where the project will be run. If it is a library that will be consumed by other applications or distributed via NuGet, the preference is usually to target .NET Standard. However, there may be APIs that are only available on .NET Core for performance or other reasons; if that's the case, .NET Core should be targeted with potentially a .NET Standard build available as well with reduced performance or functionality. By targeting .NET Standard, the project will be ready to run on new platforms (such as WebAssembly). If the project has dependencies on specific app frameworks (such as ASP.NET Core), then the target will be limited by what the dependencies support.
+When porting your application to .NET, consider the following suggestions.
```
Comment (BillWagner) [inline docs/core/porting/index.md:137]: Consider re-ordering this list so the DO recommendations are all first. That way, readers see the most important tasks first.

File: docs/core/porting/libraries.md [hunk 2]
```diff
@@ -1,105 +1,113 @@
...
 ## Retarget to .NET Framework 4.7.2
 
 If your code isn't targeting .NET Framework 4.7.2, we recommended that you retarget to .NET Framework 4.7.2. This ensures the availability of the latest API alternatives for cases where .NET Standard doesn't support existing APIs.
 
 For each of the projects you wish to port, do the following in Visual Studio:
 
-1. Right-click on the project and select **Properties**.
-1. In the **Target Framework** dropdown, select **.NET Framework 4.7.2**.
-1. Recompile the project.
+01. Right-click on the project and select **Properties**.
```
Comment (BillWagner) [inline docs/core/porting/libraries.md:41]: Curious on this: What's the difference between *01* and *1*?

File: docs/core/porting/third-party-deps.md [hunk 3]
```diff
@@ -42,21 +44,20 @@ The easiest way to inspect NuGet package folders is to use the [NuGet Package Ex
...
-Look for a folder with names using one the following patterns: `netstandardX.Y` or `netcoreappX.Y`.
+Look for a folder with names using one the following patterns: `netstandardX.Y`, `netX.Y`, or `netcoreappX.Y`.
 
-These values are the [Target Framework Monikers (TFMs)](../../standard/frameworks.md) that map to versions of [.NET Standard](../../standard/net-standard.md), .NET Core, and traditional Portable Class Library (PCL) profiles that are compatible with .NET Core.
+These values are the [Target Framework Monikers (TFMs)](../../standard/frameworks.md) that map to versions of [.NET Standard](../../standard/net-standard.md), .NET, and .NET Core, which are all compatible with .NET.
 
 > [!IMPORTANT]
-> When looking at the TFMs that a package supports, note that `netcoreapp*`, while compatible, is for .NET Core projects only and not for .NET Standard projects.
-> A library that only targets `netcoreapp*` and not `netstandard*` can only be consumed by other .NET Core apps.
+> When looking at the TFMs that a package supports, note that a TFM other than `netstandard*` targets a specific implementation of .NET, such as .NET 5, .NET Core, or .NET Framework.
```
Comment (BillWagner) [inline docs/core/porting/third-party-deps.md:52]: Do you want to add a sentence here in that ".net*" effectively replaces "netstandard*" for libraries in the .net5 + world?

**Review by BillWagner — APPROVED** [review]
Overall, this is great @adegeo 

I had a few suggestions and questions as read everything. Once you look at those, you can :shipit:

File: docs/core/porting/libraries.md (hunk 2 above)
Comment (adegeo) [reply docs/core/porting/libraries.md:41]: @BillWagner it aligns the first non-whitespace after the `.`, to the 4th character, easy for tabbing child content so they align correctly: ([commonmark spec dictates this rule](https://spec.commonmark.org/0.29/#list-items))

```markdown
1. This is 3rd column
    This is 4th column and technically wrong though the parser may ignore it for now
```
You could just use 3rd level indentation on child items, but I find this annoying when pasting things like code as a list item child

```markdown
1. This is 3rd column
   This is 3rd column and correct
```

You could use two spaces on the first to align correctly if you wanted

```markdown
1.  This is 4th column
    This is 4th column and correct, though I find two spaces in the parent line item easy to accidentally destroy
```

I find it easier to just use 01

```markdown
01. This is 4th column
    This is 4th column and correct
```

File: docs/core/porting/third-party-deps.md (hunk 3 above)
Comment (adegeo) [reply docs/core/porting/third-party-deps.md:52]: Great point!

Comment (adegeo) [reply docs/core/porting/third-party-deps.md:52]:
```suggestion
> When looking at the TFMs that a package supports, note that a TFM other than `netstandard*` targets a specific implementation of .NET, such as .NET 5, .NET Core, or .NET Framework. Starting with .NET 5, the `net*` TFM (without an operating system) effectively replaces `netstandard*` as a [portable target](../../standard/net-standard.md#net-5-and-net-standard). For example, `net5.0` targets the .NET 5 API surface and is portable and cross-platform friendly, but `net5.0-windows` targets the .NET 5 API surface as it's implemented on the Windows operating system.
```

File: docs/core/porting/index.md (hunk 1 above)
Comment (BillWagner) [reply docs/core/porting/index.md:137]: I missed your ordering. It does make sense. So, maybe add a sentence that the list is in the order to perform your tasks?

**Pull request merged**